Sources/MapboxMaps/Foundation/Camera/CameraManager.swift

```diff
@@ -525,142 +525,56 @@ public class CameraManager {
-            centerAnimationLatitude.calculationMode       = .cubic
-            centerAnimationLatitude.isRemovedOnCompletion = false
-            centerAnimationLatitude.fillMode              = .forwards
-
-            let centerAnimationLongitude = CAKeyframeAnimation(keyPath: "centerCoordinateLongitude")
-            centerAnimationLongitude.keyTimes              = keyTimes
-            centerAnimationLongitude.values                = longitudes
-            centerAnimationLongitude.duration              = time
-            centerAnimationLongitude.isAdditive            = false
-            centerAnimationLongitude.calculationMode       = .cubic
-            centerAnimationLongitude.isRemovedOnCompletion = false
-            centerAnimationLongitude.fillMode              = .forwards
-
-            animations.append(contentsOf: [centerAnimationLatitude, centerAnimationLongitude])
-        }
-
-        if camera.bearing != nil {
-            // Note - these are NOT using CAKeyframeAnimation
-            let bearingAnimation = CABasicAnimation(keyPath: "bearing")
-            bearingAnimation.toValue               = flyTo.destBearing
-            bearingAnimation.duration              = time
-            bearingAnimation.beginTime             = 0.0
-            bearingAnimation.isRemovedOnCompletion = false
-            bearingAnimation.fillMode              = .forwards
-
-            animations.append(bearingAnimation)
-        }
-
-        if camera.pitch != nil {
-            let pitchAnimation = CABasicAnimation(keyPath: "pitch")
-            pitchAnimation.toValue               = flyTo.destPitch
-            pitchAnimation.duration              = time
-            pitchAnimation.beginTime             = 0.0
-            pitchAnimation.isRemovedOnCompletion = false
-            pitchAnimation.fillMode              = .forwards
-
-            animations.append(pitchAnimation)
         }
 
-        let animationKey = String( camera.hashValue)
-        let cameraLayer = mapView.cameraView.layer
-        let animationGroup = MapboxAnimationGroup()
-        animationGroup.duration       = time
-        animationGroup.animations     = animations
-        animationGroup.fillMode       = .forwards
-        animationGroup.timingFunction = timingFunction
-
-        /// Setting `isRemovedOnCompletion` to `true` causes
-        /// the camera to reset at the end of the animation.
-        animationGroup.isRemovedOnCompletion = false
-
-        /// Remove the animation group once the animation is done.
-        animationGroup.completionBlock = { [weak cameraLayer] _ in
-            cameraLayer?.removeAnimation(forKey: animationKey)
-
-            // Temp?
-            self.setCamera(to: camera, completion: completion)
+        if let completion = completion {
+            animator.addCompletion(completion)
         }
 
-        animationGroup.delegate = animationGroup
-        cameraLayer.add(animationGroup, forKey: animationKey)
+        return animator
```

Comment:
we should call `animator.startAnimation()` before returning the `animator`.

Reply:
If we do this, then what if someone gets the animator and then runs `startAnimation()` again?

Reply:
it won't take effect

Reply:
got it